File: src/main/java/pl/lcc/yasmart/common/rtype/RewardType.java
package pl.lcc.yasmart.common.rtype;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pl.lcc.yasmart.common.account.Account;
import jakarta.validation.constraints.Email;

import java.util.UUID;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class RewardType {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    private String iconName;

    private String colorHex;

    @ManyToOne
    private Account owner;

}

File: front/src/app/domain/reward-type.ts
export class RewardType {

    constructor(public id: string|null, public name: string, public iconName:string, public colorHex: string) {
        this.name = name;
        this.iconName = iconName;
        this.colorHex = colorHex;
        this.id = id;
    }
}
